content/git/concepts/reset/reset.md

```diff
@@ -0,0 +1,88 @@
+---
+Title: 'Reset'
+Description: 'In Git, reset is used to set the HEAD pointer back to a previous commit and optionally undo staged changes and the working tree.'
+Subjects:
+  - 'Bash/Shell'
+  - 'Developer Tools'
+Tags:
+  - 'Git'
+  - 'Version Control'
+CatalogContent:
+  - 'learn-the-command-line'
+  - 'learn-git'
+---
+
+In Git, the `reset` command is used to change the state of the Git repository or undo commits.
+
+## Syntax
+
+In the terminal, the "git reset" command can be used by typing the following:
+
+```pseudo
+git reset <mode-option> <commit-reference>
+```
+
+**Note:** HEAD refers to the current commit while index refers to the staging area where files are [added using Git](https://www.codecademy.com/resources/docs/git/add).
+
+### Mode Options
+
+The `<mode-options>` refer to how far `reset` will go when rolling back changes to a previous commit, including:
+
+- Where the `HEAD` is pointing towards (usually done with just `git reset`).
+- Whether the staging area or Index, reflects the commit the `HEAD` is now pointing towards.
+- Whether the working tree is rolled back to reflect the changes reset in the `HEAD` and staging area.
+
+More specifically, these options include:
+
+- `--soft`: This rolls back to an earlier commit by moving the `HEAD` pointer towards it but leaving the staging area and working tree untouched, allowing for new commits to be made (runs by default).
+- `--mixed`: In addition to moving the `HEAD` pointer to an earlier commit, the staging area is cleared to reflect the changes made in that commit (this option runs by default).
+- `--hard`: This goes one step further and resets the working tree to reflect the previous commit reflected in the staging area and the `HEAD` pointer.
+
+The commit can either be the commit hash generated by Git, or in reference to the HEAD.
+
+## Example
+
+Suppose an error was made in a file named **example.txt** and was accidentally added and committed to a Git repository, `git reset` can be used to go back to the state before that commit was made. The default `--mixed` mode resets the `HEAD` and staging area back to the previous commit.
+
+
+This is what the terminal would look like after creating a new commit on the `main` branch and running [a `git status` check](https://www.codecademy.com/resources/docs/git/status):
+
+```
+On branch main
+Your branch is up to date with 'origin/main'.
+nothing to commit, working tree clean
+```
+
+The text above indicates the following:
+- The branch is up to date, with the `HEAD` pointing the the most recent commit.
+- There is nothing to commit in the staging area.
+- New changes haven't been made yet in the working tree.
+
+To reset the `HEAD` pointer and the staging area back by 1 commit, one of the following commands can be run:
+
+```
+git reset --hard <commit-hash>
+git reset --mixed HEAD~1
+```
+
+To receive the commit hash, run the following command:
+
+```
+git rev-parse HEAD
+```
```

Comment:
I suggest omitting this chunk only because it introduces a new term/command (`rev-parse`) that doesn't lend much to the example. We don't really need to know the actual commit hash for `HEAD`, right? We can be assured that `git reset` successfully moved the `HEAD` pointer and, thus, changed the commit hash.